src/sources/kubernetes/mod.rs

```diff
@@ -105,48 +119,161 @@ impl TimeFilter {
-    // NOTE: for now exclude images with name vector, it's a rough solution, but necessary for now
-    config
-        .exclude
-        .push((LOG_DIRECTORY.to_owned() + r"*/vector*").into());
+    file_source_include(config, &mut file_config)?;
+    file_source_exclude(config, &mut file_config);
 
-    config
-        .include
-        .push((LOG_DIRECTORY.to_owned() + r"*/*/*.log").into());
-
-    config.start_at_beginning = true;
+    file_config.start_at_beginning = true;
 
     // Filter out files that certainly don't have logs newer than now timestamp.
-    config.ignore_older = Some(10);
+    file_config.ignore_older = Some(10);
 
     // oldest_first false, having all pods equaly serviced is of greater importance
     //                     than having time order guarantee.
 
     // CRI standard ensures unique naming.
-    config.fingerprinting = FingerprintingConfig::DevInode;
+    file_config.fingerprinting = FingerprintingConfig::DevInode;
 
     // Have a subdirectory for this source to avoid collision of naming its file source.
-    config.data_dir = Some(globals.resolve_and_make_data_subdir(None, kube_name)?);
+    file_config.data_dir = Some(globals.resolve_and_make_data_subdir(None, kube_name)?);
 
     let (file_send, file_recv) = mpsc::channel(1000);
-    let file_source = config
+    let file_source = file_config
         .build("file_source", globals, file_send)
         .map_err(|e| format!("Failed in creating file source with error: {:?}", e))?;
 
     Ok((file_recv, file_source))
 }
 
+fn file_source_include(
+    config: &KubernetesConfig,
+    file_config: &mut FileConfig,
+) -> crate::Result<()> {
+    // Default includes
+    let empty = ["".to_string()];
+    let any = ["*".to_string()];
+    let uid_glob = to_uid_forms("");
+
+    // Cross product of includes
+    for object_uid in flatten_include(
+        &Some(prepare_pod_uids(&config.include_pod_uids)?),
+        uid_glob.as_slice(),
+    ) {
+        for container_name in flatten_include(&config.include_container_names, &empty) {
+            // Will include logs of Kubernetes < v1.14.
+            file_config.include.push(
+                (LOG_DIRECTORY.to_owned()
+                    + format!("{}/{}*/*.log", object_uid, container_name).as_str())
+                .into(),
+            );
+
+            for namespace in flatten_include(&config.include_namespaces, &any) {
+                // Will include logs of Kubernetes >= v1.14.
+                file_config.include.push(
+                    (LOG_DIRECTORY.to_owned()
+                        + format!("{}_*_{}/{}*/*.log", namespace, object_uid, container_name)
+                            .as_str())
+                    .into(),
+                );
+            }
+        }
+    }
+
+    Ok(())
+}
+
+// Parses partial UIDs and transforms them to two UID forms with glob [0-9A-Fa-f] filling
+// them up to 32 hexadecimal numbers.
```

Comment:
can you expand up on these docs, its not directly clear to me what this achieves.